Assets/Scripts/ssjAug2018/World/Mailbox/MailboxManager.cs

```diff
@@ -109,9 +82,37 @@ public void ActivateMailboxGroup()
                 letterCount = (letterCount > setSize) ? setSize : letterCount;
                 
                 box.ActivateMailbox(letterCount);
+                _activeMailboxes++;
                 validBoxes.Remove(box);
                 setSize =- letterCount;
             }
         }
+
+        private List<Mailbox> GetMailboxesInRange(Transform origin, float minimum, float maximum)
+        {
+            List<Mailbox> validBoxes = new List<Mailbox>();
+            Collider[] hits = new Collider[_mailboxes.Count];
```

Comment:
Well, you don't wanna allocate here either. You should allocate a reasonably sized buffer in Awake() and then re-use it. The idea being we don't want to do any allocations while the game is running.

Reply:
Is `_mailboxes.Count` unreasonable? It's max number that could happen, which is most likely at the start of the game (Unless there's some close enough to not be found.)

I suppose it could also be `_mailboxes.Count - (_mailboxes.Count * 0.8)` or some such percentage to ensure there's enough room for a variety of the boxes without always being the max size.

Reply:
Fuck. Scratch all that. That's just going to be 0 since none of the boxes are registered until after it has hit `Awake()`
I'll figure something out.

Reply:
Thoughts on this: Can we allocate it during the loading stage? Wait for the boxes to register themselves, then allocate its size?

Reply:
I added an event to GameManager called GameReadyEvent or something. that'll get called when the main scene is all loaded and ready, so you should be able to do it there (verify that first tho),